src/main.ts

```diff
@@ -135,6 +138,25 @@ class TimeControls {
       this.pauseBtn.disabled = false;
     }
   }
+
+  /**
+   * Sets the current frame and updates this TimeControls UI to match.
+   * @returns true if the frame was set correctly (false if the frame is out of range).
+   */
+  public setCurrentFrame(frame: number): boolean {
+    if (this.goToFrame(frame)) {
+      this.currentFrame = frame;
```

Comment:
this.goToFrame does actually set this.currentFrame so that line is not needed.

Reply:
Interestingly, this function looks a lot like handleFrameAdvance (except it takes an absolute number instead of a delta).  Maybe could just reuse that function! handleFrameAdvance(newframe-currentFrame)

Reply:
or `handleFrameAdvance` could just call this function but convert from relative to absolute numbers first?

Reply:
Good catch, thank you both!